### PR TITLE
Add remote partitioning samples

### PR DESCRIPTION
**Pull request by benas**
This PR fixes the code examples in the remote partitioning section of the documentation and adds two remote partitioning samples:

* One that shows how to poll the job repository to monitor workers progress
* And another one that shows how to aggregate replies sent by workers to notify work completion

The sample deliberately uses a simple worker step (just a tasklet that prints to the console the name of the partition it is processing) in order to focus on how to configure a remote partitioning job.

An embedded JMS broker and an embedded database are used for simplicity.

This PR resolves [BATCH-2730](https://jira.spring.io/browse/BATCH-2730) and serves as a base for [BATCH-2687](https://jira.spring.io/browse/BATCH-2687).

### CODE REVIEW

File: spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithRepositoryPollingFunctionalTests.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * The master step of the job under test will create 3 partitions for workers
+ * to process.
+ *
+ * @author Mahmoud Ben Hassine
+ */
+@ContextConfiguration(classes = {JobRunnerConfiguration.class, MasterConfiguration.class})
+public class RemotePartitioningJobWithRepositoryPollingFunctionalTests extends RemotePartitioningJobFunctionalTests {
```
Comment (mminella) [inline spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithRepositoryPollingFunctionalTests.java]: Unless I'm mistaken, the only difference between the two functional tests is the imports (and the names).  Can we look at either using JUnit's parameterized tests or at a minimum, putting everything except how the `ApplicationContext` is created in the parent and just have the subclasses provide their `ApplicationContext` with the correct bits (so the parent does the test itself as well).

File: spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithMessageAggregationFunctionalTests.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * The master step of the job under test will create 3 partitions for workers
+ * to process.
+ *
+ * @author Mahmoud Ben Hassine
+ */
+@ContextConfiguration(classes = {JobRunnerConfiguration.class, MasterConfiguration.class})
+public class RemotePartitioningJobWithMessageAggregationFunctionalTests extends RemotePartitioningJobFunctionalTests {
```
Comment (mminella) [inline spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithMessageAggregationFunctionalTests.java]: Unless I'm mistaken, the only difference between the two functional tests is the imports (and the names). Can we look at either using JUnit's parameterized tests or at a minimum, putting everything except how the ApplicationContext is created in the parent and just have the subclasses provide their ApplicationContext with the correct bits (so the parent does the test itself as well).

File: spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithRepositoryPollingFunctionalTests.java (hunk 1 above)
Comment (benas) [reply spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithRepositoryPollingFunctionalTests.java]: Good catch! I updated the PR accordingly

File: spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithMessageAggregationFunctionalTests.java (hunk 2 above)
Comment (benas) [reply spring-batch-samples/src/test/java/org/springframework/batch/sample/RemotePartitioningJobWithMessageAggregationFunctionalTests.java]: Good catch! I updated the PR accordingly

File: spring-batch-samples/README.md [hunk 3]
```diff
@@ -638,13 +638,25 @@ happens inside the main step transaction).
...
+
+This sample shows how to configure a remote partitioning job. The master step
+uses a `MessageChannelPartitionHandler` to send partitions to and receive 
+replies from workers. Two examples are shown:
+
+* A master step that polls the job repository to see if all workers have finished
+their work
+* A master step that aggregates replies from workers to notify work completion
+
+The sample uses an embedded JMS broker and an embedded database for simplicity.
```
[the remark this answers is not in the document]
Comment (benas) [reply spring-batch-samples/README.md]: That's better. I updated the README.

**Comment by mminella** [comment]
Rebased, squashed, and merged as 5730f380e28a822681b6685a69695ee54a0edec6.

**Pull request closed without merging**